python/cudf/cudf/core/index.py

```diff
@@ -1410,6 +1444,113 @@ def where(self, cond, other=None, inplace=False):
             inplace=inplace,
         )
 
+    @property
+    def values(self):
+        return self._column.values
+
+    def __contains__(self, item):
+        return item in self._values
+
+    def _clean_nulls_from_index(self):
+        """
+        Convert all na values(if any) in Index object
+        to `<NA>` as a preprocessing step to `__repr__` methods.
+
+        This will involve changing type of Index object
+        to StringIndex but it is the responsibility of the `__repr__`
+        methods using this method to replace or handle representation
+        of the actual types correctly.
+        """
+        if self._values.has_nulls():
+            return cudf.Index(
+                self._values.astype("str").fillna(cudf._NA_REP), name=self.name
+            )
+
+        return self
+
+    def to_frame(self, index=True, name=None):
+        if name is not None:
+            col_name = name
+        elif self.name is None:
+            col_name = 0
+        else:
+            col_name = self.name
+        return cudf.DataFrame(
+            {col_name: self._values}, index=self if index else None
+        )
+
+    def any(self):
+        if len(self) == 0:
+            return False
+
+    def to_pandas(self):
+        return pd.Index(self._values.to_pandas(), name=self.name)
+
+    def append(self, other):
+        if is_list_like(other):
+            to_concat = [self]
+            to_concat.extend(other)
+        else:
+            this = self
+            if len(other) == 0:
+                # short-circuit and return a copy
+                to_concat = [self]
+
+            other = cudf.Index(other)
+
+            if len(self) == 0:
+                to_concat = [other]
+
+            if len(self) and len(other):
+                if is_mixed_with_object_dtype(this, other):
+                    got_dtype = (
+                        other.dtype
+                        if this.dtype == cudf.dtype("object")
+                        else this.dtype
+                    )
+                    raise TypeError(
+                        f"cudf does not support appending an Index of "
+                        f"dtype `{cudf.dtype('object')}` with an Index "
+                        f"of dtype `{got_dtype}`, please type-cast "
+                        f"either one of them to same dtypes."
+                    )
+
+                if isinstance(self._values, cudf.core.column.NumericalColumn):
+                    if self.dtype != other.dtype:
+                        this, other = numeric_normalize_types(self, other)
+                to_concat = [this, other]
+
+        for obj in to_concat:
+            if not isinstance(obj, BaseIndex):
```

Comment:
(Not sure if this code is just moved or newly added)

Maybe more straightforward if `if not isinstance(other, BaseIndex)` at the top (we always know `to_concat[0]` should be a BaseIndex here)

Reply:
does this mean we don't wanna consider case where `other `is `is_list_like`?

Reply:
Ah you're correct, I didn't realize `other` could be a list with multiple items.